independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Types.java

```diff
@@ -164,13 +165,16 @@ static Type getProviderType(ClassInfo classInfo) {
     static Set<Type> getProducerMethodTypeClosure(MethodInfo producerMethod, BeanDeployment beanDeployment) {
         Set<Type> types;
         Type returnType = producerMethod.returnType();
+        if (returnType.kind() == Kind.TYPE_VARIABLE) {
+            throw new DefinitionException("A type variable is not a legal bean type: " + producerMethod);
```

Comment:
Perhaps this can be improved? I'm bringing it up because I don't even remember what a type variable is :P

Reply:
> I don't even remember what a type variable is...

I will pretend that you've never written this ;-).

Type variables are type parameters, e.g.:
```
class ProducingBean<T> {

   // This is illegal...
   @Produces
   T produce() {
    // ...
   }
}
```

Reply:
Hahaha

I meant that just mentioning type parameter as a Jandex term is unlikely to help a user understand what's going on. Perhaps just mention generics - that way everyone will know what the problem is

Reply:
Hm, the message contains the declaring class, method signature and return type... and it's a definition error as defined [by the spec](https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#producer_method). I'd keep it simple for now. It's not a common use case...

Reply:
OK :)